web_monitoring/internetarchive.py

```diff
@@ -28,90 +31,157 @@ class UnexpectedResponseFormat(WebMonitoringException):
     ...
 
 
-TIMEMAP_URL_TEMPLATE = 'http://web.archive.org/web/timemap/link/{}'
-DATE_FMT = '%a, %d %b %Y %H:%M:%S %Z'
-DATE_URL_FMT = '%Y%m%d%H%M%S'
-URL_CHUNK_PATTERN = re.compile('\<(.*)\>')
-DATETIME_CHUNK_PATTERN = re.compile(' datetime="(.*)",')
+CDX_SEARCH_URL = 'http://web.archive.org/cdx/search/cdx'
+ARCHIVE_RAW_URL_TEMPLATE = 'http://web.archive.org/web/{timestamp}id_/{url}'
+ARCHIVE_VIEW_URL_TEMPLATE = 'http://web.archive.org/web/{timestamp}/{url}'
+URL_DATE_FORMAT = '%Y%m%d%H%M%S'
 
+CdxRecord = namedtuple('CdxRecord', (
+    # Raw CDX values
+    'key',
+    'timestamp',
+    'url',
+    'mime_type',
+    'status_code',
+    'digest',
+    'length',
+    # Synthesized values
+    'date',
+    'raw_url',
+    'view_url'
+))
 
-def list_versions(url):
+
+def cdx_hash(content):
+    if isinstance(content, str):
+        content = content.encode()
+    return b32encode(hashlib.sha1(content).digest())
+
+
+def search_cdx(params):
     """
-    Yield (version_datetime, version_uri) for all versions of a url.
+    Search archive.org's CDX API for captures of a given URL. This will
+    automatically page through all results for a given search.
+
+    Returns an iterator of CdxRecord objects. The StopIteration value is the
+    total count of found captures.
+
+    Note that even URLs without wildcards may return results with different
+    URLs. Search results are matched by url_key, which is a SURT-formatted,
+    canonicalized URL:
```

Comment:
Ah, is this Numpy’s style? https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt